test/ZNSStringResolver.test.ts

```diff
@@ -0,0 +1,433 @@
+    });
+
+    it("Should setAccessController() correctly with ADMIN_ROLE", async () => {
+      await expect(
+        stringResolver.connect(admin).setAccessController(admin.address)
+      ).to.emit(
+        stringResolver, "AccessControllerSet"
+      ).withArgs(admin.address);
+
+      expect(
+        await stringResolver.getAccessController()
+      ).to.equal(admin.address);
+    });
+
+    it("Should revert when setAccessController() without ADMIN_ROLE", async () => {
+      await expect(
+        stringResolver.connect(user).setAccessController(user.address)
+      ).to.be.revertedWith(
+        getAccessRevertMsg(user.address, ADMIN_ROLE)
+      );
+    });
+
+    it("Should support the IZNSAddressResolver interface ID", async () => {
+      const interfaceId = await stringResolver.getInterfaceId();
+      const supported = await stringResolver.supportsInterface(interfaceId);
+      expect(supported).to.be.true;
+    });
+
+    it("Should support the ERC-165 interface ID", async () => {
+      expect(
+        await stringResolver.supportsInterface(
+          ERC165__factory.createInterface()
+            .getFunction("supportsInterface").selector
+        )
+      ).to.be.true;
+    });
+
+    it("Should not support other interface IDs", async () => {
+      expect(
+        await stringResolver.supportsInterface("0xffffffff")
+      ).to.be.false;
+    });
+
+
+    describe("UUPS", () => {
+
+      it("Allows an authorized user to upgrade the StringResolver", async () => {
+        // deployer deployed, deployAdmin wanna edit
+        const factory = new ZNSStringResolverUpgradeMock__factory(deployer);
+        const newStringResolver = await factory.deploy();
+        await newStringResolver.waitForDeployment();
+
+        // Confirm the deployer is a governor
+        expect(
+          await accessController.hasRole(GOVERNOR_ROLE, deployAdmin.address)
+        ).to.be.true;
+
+        const upgradeTx = domainToken.connect(deployAdmin).upgradeTo(await newStringResolver.getAddress());
+
+        await expect(upgradeTx).to.not.be.reverted;
+      });
+
+      it("Fails to upgrade if the caller is not authorized", async () => {
+        const factory = new ZNSStringResolverUpgradeMock__factory(deployer);
+
+        // DomainToken to upgrade to
+        const newStringResolver = await factory.deploy();
+        await newStringResolver.waitForDeployment();
+
+        // Confirm the operator is not a governor
+        await expect(
+          accessController.checkGovernor(operator.address)
+        ).to.be.revertedWith(
+          getAccessRevertMsg(operator.address, GOVERNOR_ROLE)
+        );
+
+        const upgradeTx = domainToken.connect(operator).upgradeTo(await newStringResolver.getAddress());
+
+        await expect(upgradeTx).to.be.revertedWith(
+          getAccessRevertMsg(operator.address, GOVERNOR_ROLE)
+        );
+      });
+
+      // TODO: Falls on the role. I think, cannot give a REGISTRAR_ROLE to mock "deployAdmin".
```

Comment:
Make sure you take care of any existing TODO comments. If they are already fixed, remove them